test/Microsoft.AspNetCore.Mvc.Routing.Abstractions.Tests/RouteValueDictionaryTests.cs

```diff
@@ -156,21 +156,6 @@ public void CreateFromObject_CopiesPropertiesFromRegularType_WithIndexerProperty
             Assert.Equal(0, dict.Count);
         }
 
-        [Fact]
-        public void CreateFromObject_MixedCaseThrows()
```

Comment:
We added this test because it could thoroughly confuse users if the code silently chose one value over the other. Or perhaps I did cause a customer issue...

Either way, we shouldn't lose this behaviour.